### PR TITLE
Change condition from empty to "empty" string

### PR DESCRIPTION
**Pull request by zonky2**
Change condition from empty to "empty" string - now it´s possible to set 0 as default value

C 4.9

### CODE REVIEW

**Comment by Aybee** [comment]
"emty"? Did you mean "empty"?

**Comment by zonky2** [comment]
ich kaufe ein "p" :D

File: core-bundle/src/Resources/contao/templates/forms/form_range.html5 [hunk 1]
```diff
@@ -17,5 +17,5 @@
     <p class="error"><?= $this->getErrorAsString() ?></p>
   <?php endif; ?>
 
-  <input type="<?= $this->type ?>" name="<?= $this->name ?>" id="ctrl_<?= $this->id ?>" class="range<?php if ($this->class): ?> <?= $this->class ?><?php endif; ?>"<?php if ($this->value): ?> value="<?= (float) $this->value ?>"<?php endif; ?><?= $this->getAttributes() ?>>
+  <input type="<?= $this->type ?>" name="<?= $this->name ?>" id="ctrl_<?= $this->id ?>" class="range<?php if ($this->class): ?> <?= $this->class ?><?php endif; ?>"<?php if ('' != $this->value): ?> value="<?= (float) $this->value ?>"<?php endif; ?><?= $this->getAttributes() ?>>
```
Comment (ausi) [inline core-bundle/src/Resources/contao/templates/forms/form_range.html5]: I think we should use strict comparison (`!==`) in this case.
```suggestion
  <input type="<?= $this->type ?>" name="<?= $this->name ?>" id="ctrl_<?= $this->id ?>" class="range<?php if ($this->class): ?> <?= $this->class ?><?php endif; ?>"<?php if ('' !== $this->value): ?> value="<?= (float) $this->value ?>"<?php endif; ?><?= $this->getAttributes() ?>>
```

Comment (leofeyer) [reply core-bundle/src/Resources/contao/templates/forms/form_range.html5]: We are casting the value to `(float)` but we use `'' !==` to compare it to an empty string. Is this intended?

Comment (zonky2) [reply core-bundle/src/Resources/contao/templates/forms/form_range.html5]: I would say this is correct - value can be an int or float and should only be set if the input field contains an entry, which can also be 0

Comment (ausi) [reply core-bundle/src/Resources/contao/templates/forms/form_range.html5]: > Is this intended?

I think so, yes. The `value` attribute should always be a number, that’s why we cast to `(float)`. The `'' !==` is used to differentiate between int/float `0` (in which case we want to output a value) and the empty string (which we do not want to output).

**Review by ausi — APPROVED** [review]

File: core-bundle/src/Resources/contao/templates/forms/form_range.html5 (hunk 1 above)
Comment (leofeyer) [reply core-bundle/src/Resources/contao/templates/forms/form_range.html5]: What if `$this->value` is `null` or `false`? The if condition would be true in this case and the attribute would be set to `value="0"`. IMHO, it should be `'' !== (string) $this->value`, shouldn't it?

Comment (zonky2) [reply core-bundle/src/Resources/contao/templates/forms/form_range.html5]: when the form widget is created $this->value is the default value - current its not possible to set 0 (zero)!

**Review by leofeyer — CHANGES_REQUESTED** [review]
See https://github.com/contao/contao/pull/2307#discussion_r517265771:



Also, the PR needs to be rebased on the `4.9` branch if this is a bugfix.

**Comment by fritzmg** [comment]
Closing in favor #2779

**Pull request closed without merging**